### ISSUE TITLE
Traffic Intersection: cars are running red lights!

### ISSUE THREAD

**Issue by nicolaspayette** [NetLogo/models#147]
Which is why, fittingly, we get this _ticket_.

But in all seriousness, this is seems to happen fairly systematically with the very first eastbound car (and sometimes the second one) after setup. I have also seen it happen later in a run (in both directions) but it's harder to pinpoint what circumstances lead to it then.

(I have noticed this while working on #140 and I may try to address it at the same time.)

**Comment by nicolaspayette** [comment]
I committed a fix (11a6b1a) in the `weed-out-wait` branch, but I also noticed that (even before my changes) the info tab of the model is not entirely truthful about its behavior. The info tab says:

> - I stop for cars in front of me and red lights, and I slow down at a yellow light.
> - If I am moving quickly and I see that I will have to stop soon, I slow down proportional to the distance of non-free space up to MAX-BRAKE.
> - If I see that I have free space in front of me, I speed up proportional to the amount of free space up to MAX-ACCEL.

Two things are wrong with that:

- The model currently ignores yellow lights entirely (i.e., it treats them as green lights).
- There is nothing "proportional" about accelerating or breaking: cars accelerate or break as hard as they can.

I think we should:

- Treat yellow lights as red lights (you're supposed to stop at a yellow light, aren't you?) It's a very simple change and wouldn't change the behavior of the model much.
- Rewrite the part about acceleration/deceleration to reflect to actual behavior of the model. I'm not even sure what "proportional to the amount of free space" is supposed to mean exactly and I think it would needlessly complicate the model to try to fit that description.
